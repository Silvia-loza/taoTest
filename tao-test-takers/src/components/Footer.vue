<template>
    <footer class="footer">
    <ul>
    <li>Products</li>
    <li>Solutions</li>
    <li>Services</li>
    <li>Support</li>
    <li>Company</li>
    </ul>
    </footer>
</template>

<script>
export default {
  name: 'Footer',
  props: {}
}
</script>

<style scoped>

.footer{
display:flex;
align-items: center;
justify-content: space-between;
background-color:#263e4e ;
color: white;
height: 30vh;
font-size: 3em
}

</style>
