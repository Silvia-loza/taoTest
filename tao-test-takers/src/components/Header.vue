<template>
  <header class="header">
   <img id=logoHeader alt="Tao Testing logo" src="../assets/TAOTesting.png">
    <ul>
      <li>Product</li>
      <li>Solutions</li>
      <li>Services</li>
      <li>Support</li>
      <li>About Us</li>
      <li>Resources</li>
      <li>Blog</li>
    </ul>
  </header>
</template>

<script>
export default {
  name: 'Header',
  props: {
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
*{
color: #bc262d;
margin: 0 auto;
box-sizing: border-box;
}

#logoHeader{
justify-content: flex-start;
width:10%;
margin-top: 2%;
margin-bottom: 2%;
}

.header {
  height: inherit;
  width: inherit;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-shadow: 0.5px 9px 9px 4px rgba(0, 0, 0, 0.2);
   }
   ul {
   width:60%;
  }
   li {
    margin: 0 3%;
    font-size: 120%;
   }
</style>
