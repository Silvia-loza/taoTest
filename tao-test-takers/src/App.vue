<template>
  <div>
   <Header/>
   <TestTaker/>
   <Footer/>
  </div>
</template>
<script>
import Header from './components/Header.vue'
import Footer from './components/Footer.vue'
import TestTaker from './components/test-taker'

export default {
  name: 'App',
  components: {
    Header,
    Footer,
    TestTaker
  }
}
</script>
<style>
*{
margin: 0 auto;
box-sizing: border-box;}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
}
</style>
