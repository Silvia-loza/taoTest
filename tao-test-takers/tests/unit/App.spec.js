import { shallowMount } from '@vue/test-utils'
import App from '@/App'
import Footer from '@/components/Footer.vue'
import Header from '@/components/Header.vue'
import TestTaker from '@/components/test-taker.vue'


describe("App works", () => {
    it("renders the component", () => {
        //arrange item to be mounted
        const wrapper = shallowMount(App)
        //assert that the component is mounted and that it matches the snapshot creted, will inform us about changes.
        expect(wrapper.html()).toMatchSnapshot()
    })
    const build = () => {
        const wrapper = shallowMount(App)
        return {
            wrapper,
            Header: () => wrapper.findComponent(Header),
            Footer: () => wrapper.findComponent(Footer),
            TestTaker: () => wrapper.findComponent(TestTaker),
        }
    }
    it("renders the right", () => {
        const { Footer, Header, TestTaker } = build()

        expect(Header().exists()).toBe(true)
        expect(Footer().exists()).toBe(true)
        expect(TestTaker().exists()).toBe(true)
    })
})



