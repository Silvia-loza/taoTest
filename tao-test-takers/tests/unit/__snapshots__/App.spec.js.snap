// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App works renders the component 1`] = `
<div>
  <header-stub></header-stub>
  <test-taker-stub></test-taker-stub>
  <footer-stub></footer-stub>
</div>
`;
